### PR TITLE
feat(profiling): add flamegraph breakdown

### PR DESCRIPTION
**Pull request by JonasBa**
Add a paginated list of top functions.

CleanShot 2024-09-25 at 14 01 15@2x

### CODE REVIEW

File: static/app/views/profiling/landingAggregateFlamegraph.tsx [hunk 1]
```diff
@@ -129,6 +168,66 @@ function AggregateFlamegraphToolbar(props: AggregateFlamegraphToolbarProps) {
...
+                      {' '}
+                      ({frame.frame.package || frame.frame.module})
+                    </FunctionPackage>
+                  )}
+                </TextOverflow>
+              </FunctionInfo>
+              <Metric>
+                {t('Samples')}: {formatAbbreviatedNumber(frame.aggregate.totalWeight)}
+                {/* {' | '}
+                {t('Instances')}: {frame.instances.length} */}
```
Comment (Zylphrex) [inline static/app/views/profiling/landingAggregateFlamegraph.tsx:205]: Did you mean to leave this here?

File: static/app/views/profiling/landingAggregateFlamegraph.tsx [hunk 2]
```diff
@@ -52,6 +60,37 @@ function decodeViewOrDefault(
...
+}
+
+function sortByTotalWeight(a: SlowestFlamegraphFrame, b: SlowestFlamegraphFrame) {
+  return b.aggregate.totalWeight - a.aggregate.totalWeight;
+}
+
+function getSlowestFlamegraphFrames(flamegraph: Flamegraph): SlowestFlamegraphFrame[] {
+  const results: Map<FlamegraphFrame['frame'], SlowestFlamegraphFrame> = new Map();
+
+  for (const frame of flamegraph.frames) {
```
Comment (Zylphrex) [inline static/app/views/profiling/landingAggregateFlamegraph.tsx:79]: I forget but does this flamegraph update to only contain application/system frames the user selects it?

**Review by Zylphrex — APPROVED** [review]
Be nice if we can bring this functionality into the profiles table under transaction summary.

File: static/app/views/profiling/landingAggregateFlamegraph.tsx (hunk 2 above)
Comment (JonasBa) [reply static/app/views/profiling/landingAggregateFlamegraph.tsx:79]: Yes it does. Tbh if we want that tbh, I could see it being useful, but then we also usually also have dedicated controls for this...

File: static/app/views/profiling/landingAggregateFlamegraph.tsx (hunk 1 above)
Comment (JonasBa) [reply static/app/views/profiling/landingAggregateFlamegraph.tsx:205]: I added it, but I'm unclear if its useful to know call counts. I will likely delete it, `instances` just seems like bad wording

File: static/app/views/profiling/landing/slowestFunctionsTable.tsx [hunk 3]
```diff
@@ -61,7 +61,7 @@ function makeProfileLinkFromExample(
   return null;
 }
 
-function useMemoryPagination(items: any[], size: number) {
+export function useMemoryPagination(items: any[], size: number) {
```
[the remark this answers is not in the document]
Comment (JonasBa) [reply static/app/views/profiling/landing/slowestFunctionsTable.tsx:64]: Yeah, I'll move it out

**Pull request closed without merging**